tests/source_storages/test_source_storages.py

```diff
@@ -1,41 +1,99 @@
-import uuid
-
 import pytest
 
-from ragna.core import LocalDocument
+from ragna.core import LocalDocument, MetadataFilter
 from ragna.source_storages import Chroma, LanceDB
 
+METADATAS = [
+    {"key": "value"},
+    {"key": "value", "other_key": "other_value"},
+    {"key": "other_value"},
+    {"other_key": "value"},
+    {"other_key": "other_value"},
+    {"key": "foo"},
+    {"key": "bar"},
+]
+
+metadata_filters = pytest.mark.parametrize(
+    "metadata_filter,n_expected_sources",
+    [
+        pytest.param(
+            MetadataFilter.and_(
+                [
+                    MetadataFilter.eq("key", "value"),
+                    MetadataFilter.eq("other_key", "other_value"),
+                ]
+            ),
+            1,
+            id="and",
+        ),
+        pytest.param(
+            MetadataFilter.or_(
+                [
+                    MetadataFilter.eq("key", "value"),
+                    MetadataFilter.eq("key", "other_value"),
+                ]
+            ),
+            3,
+            id="or",
+        ),
+        pytest.param(
+            MetadataFilter.and_(
+                [
+                    MetadataFilter.eq("key", "value"),
+                    MetadataFilter.or_(
+                        [
+                            MetadataFilter.eq("key", "other_value"),
+                            MetadataFilter.ne("other_key", "other_value"),
+                        ]
+                    ),
+                ]
+            ),
+            1,
+            id="and-nested",
+        ),
+        pytest.param(
+            MetadataFilter.or_(
+                [
+                    MetadataFilter.eq("key", "value"),
+                    MetadataFilter.and_(
+                        [
+                            MetadataFilter.eq("key", "other_value"),
+                            MetadataFilter.ne("other_key", "other_value"),
+                        ]
+                    ),
+                ]
+            ),
+            2,
+            id="or-nested",
+        ),
+        pytest.param(MetadataFilter.eq("key", "value"), 2, id="eq"),
+        pytest.param(MetadataFilter.ne("key", "value"), 5, id="ne"),
+        pytest.param(MetadataFilter.in_("key", ["foo", "bar"]), 2, id="in"),
+        pytest.param(MetadataFilter.not_in("key", ["foo", "bar"]), 5, id="not_in"),
+        pytest.param(None, 7, id="none"),
```

Comment:
@pmeier @nenb I need to investigate why some of these tests are failing. Chroma and LanceDB return different sources for the same MetadataFilters over the same documents.